index.bs

```diff
@@ -438,32 +431,119 @@ partial interface NeuralNetworkContext {
+                - output tensor: [batches, channels, height, width]
+            "nhwc":
+                - input tensor: [batches, height, width, channels]
+                - output tensor: [batches, height, width, channels]
+
+    **Returns:** an {{Operand}}. The output 4-D tensor that contains the
+    result of the reduction. The logical shape is interpreted according to the
+    value of *layout*.
+
+    <div class="note">
+    A "global" pooling operation such as the *globalMaxPool2d* operation is a variant of the pooling operation where the window dimensions is the dimensions of the input as follow.
+    <pre highlight="js">
+    Operand globalMaxPool2d(input) {
+      return maxPool2d(input, input.shape());
+    }
+    </pre>
+    </div>
+</div>
+
+### reduce ### {#api-neuralnetworkcontext-reduce}
+Reduce the input along the dimensions given in *axes*.
+<script type=idl>
+partial interface NeuralNetworkContext {
+  Operand reduceL1(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+  Operand reduceL2(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+  Operand reduceLogSum(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+  Operand reduceLogSumExp(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+  Operand reduceMax(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+  Operand reduceMean(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+  Operand reduceMin(Operand input, optional sequence<long> axe = null, optional boolean keepDimensions = false);
+  Operand reduceProduct(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+  Operand reduceSum(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+  Operand reduceSumSquare(Operand input, optional sequence<long> axes = null, optional boolean keepDimensions = false);
+};
+</script>
+<div algorithm=reduce>
+    **Arguments:**
+        - *input*: an {{Operand}}. The input tensor.
+        - *axes*: an optional sequence of {{long}}. The dimensions to reduce.
+            If not present, all dimensions are reduced.
+        - *keepDimensions*: an optional {{boolean}}. If true, retains reduced dimensions with size of 1.
+            The default value is false.
+
+    **Returns:** an {{Operand}}. The reduced output tensor.
+
+    **Reduction types:**
+        - *L1*: Compute the <a href="https://mathworld.wolfram.com/L1-Norm.html">L1 norm</a> of the input elements along the axes.
+        - *L2*: Compute the <a href="https://mathworld.wolfram.com/L2-Norm.html">L2 norm</a> of the input elements along the axes.
+        - *LogSum*: Compute the `log(sum(input values))` of the input values along the axes.
```

Comment:
"input elements" is used for L1 and L2. "input values" is used for others. Would it be better to be consistent?

Reply:
Fixed.